x/vote/keeper/migrate.go

```diff
@@ -179,6 +183,39 @@ func deleteAllPendingPolls(ctx sdk.Context, k Keeper) {
 }
 
 func migrateAllCompletedPolls(ctx sdk.Context, k Keeper) {
+	pollMetadatas := k.getCompletedPollMetadatas(ctx)
+
+	var voterKeys []utils.Key
+	var voterRecords []types.VoteRecord
+	var pollKeys []exported.PollKey
+
+	for _, p := range pollMetadatas {
+		poll := k.newPollStore(ctx, p.Key)
+		voterIter := k.getKVStore(ctx).Iterator(voterPrefix.AppendStr(poll.key.String()))
+		for ; voterIter.Valid(); voterIter.Next() {
+			voterKeys = append(voterKeys, voterIter.GetKey())
+			voterRecords = append(voterRecords, types.VoteRecord{IsLate: false})
```

Comment:
What was the motivation for rearranging this? This seems like it would be less efficient since we're looping over multiple times and storing auxiliary data?

Reply:
That makes sense, thanks